### PR TITLE
Form

### PR DESCRIPTION
**Pull request by eduardojmatos**
**CHANGELOG** :memo:

* Now we have a form on Gaiden! Some components come with form, like:
  * `form__field` - to encapsulate widget elements;
  * `form__input` - input, select, textarea, etc.;
  * `form__label` - captain obvious;
  * `form__message` - with invalid message;
  * `form__radio` - personalized radio button;
  * `form__check` - checkbox input;
  * `form__steps` - to footer steps on form;
* Minor adjust on widget (using box-shadow instead border);

**PRINTS** 🖼 

screencapture-localhost-8000-demo-gaiden-css-components-form-demo-html-1481240277683

### CODE REVIEW

File: src/scss/components/_form.scss [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+**/
+
+.form {
+
+  &__container {
+    padding: {
+      right: 10%;
+      left: 10%;
+    }
+    width: 100%;
```
Comment (GeorgeJegon) [inline src/scss/components/_form.scss]: Precisa disso aqui mesmo? ele já não vai ter o tamanho de `100%` ou `auto`

File: src/scss/components/_form.scss [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    & + & {
+      margin-top: 16px;
+    }
+  }
+
+  &__input {
+    display: block;
+    padding: 13px;
+    border: 1px solid get-color((color: punch, opacity: .5));
+    border-radius: 6px;
```
Comment (GeorgeJegon) [inline src/scss/components/_form.scss]: We could transform `6px` for border-radius in a variable, could we?

File: src/scss/components/form/_checkbox.scss [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  input {
+    position: relative;
+    z-index: 2;
+  }
+
+  > input {
+    width: 100%;
+    height: 100%;
+    position: absolute;
+    left: 0;
```
Comment (GeorgeJegon) [inline src/scss/components/form/_checkbox.scss]: I think that when `position` is absolute the `left` and `top` rules are 0.

File: src/scss/components/_form.scss (hunk 2 above)
Comment (eduardojmatos) [inline src/scss/components/_form.scss]: @GeorgeJegon we have a bunch of border-radius on all styleguide and designers still be undecided about which are the ideal border-radius to application, or if they will have more than one value to this. It is better wait this decision to make all this values a map or a variable.

*maybe use a `get-radius(input)` 😄

File: src/scss/components/_form.scss (hunk 1 above)
Comment (eduardojmatos) [inline src/scss/components/_form.scss]: @GeorgeJegon fieldset, by default, is blocked. My concern about this is a case where someone will use this `form__container` on another tag, and not in fieldset.

Maybe we can rename `form__container` to `form__fieldset`. What do you think?

Comment (GeorgeJegon) [inline src/scss/components/_form.scss]: Hmmm got it. Makes sense, ensure that this element always will be fluid 👍

File: src/scss/components/_form.scss (hunk 2 above)
Comment (GeorgeJegon) [inline src/scss/components/_form.scss]: Yeah!!!!

**Pull request merged**